pallets/runtime/tests/src/utility_test.rs

```diff
@@ -228,3 +231,91 @@ fn _relay_unhappy_cases() {
         Error::InvalidNonce
     );
 }
+
+#[test]
+fn batch_secondary_with_permissions_works() {
+    ExtBuilder::default()
+        .build()
+        .execute_with(batch_secondary_with_permissions);
+}
+
+fn batch_secondary_with_permissions() {
+    System::set_block_number(1);
+    let alice_key = AccountKeyring::Alice.public();
+    let alice_origin = Origin::signed(alice_key);
+    let alice_did = register_keyring_account_with_balance(AccountKeyring::Alice, 1_000).unwrap();
+    let bob_key = AccountKeyring::Bob.public();
+    let bob_origin = Origin::signed(bob_key);
+    let bob_signer = Signatory::Account(bob_key);
```

Comment:
This can be cleaned up a bit using the `User` abstraction in `pips_test`.

Reply:
OK, however note that Bob doesn't have a DID and `User` assigns a DID by default.